### PR TITLE
Adds intercepting logger to handle multiple sinks

### PR DESCRIPTION
**Pull request by drewbailey**
Adds InterceptingLogger interface as a way to emit log messages to multiple sinks while allowing for independent log levels of each sink and root logger.

### CODE REVIEW

File: logger.go [hunk 1]
```diff
@@ -126,9 +126,14 @@ type Logger interface {
 	// Indicate if ERROR logs would be emitted. This and the other Is* guards
 	IsError() bool
 
+	ImpliedArgs() []interface{}
```
Comment (schmichael) [inline logger.go]: Comment
```suggestion
	// ImpliedArgs returns With key/value pairs
	ImpliedArgs() []interface{}
```

Comment (drewbailey) [reply logger.go]: thanks!

File: logger_test.go [hunk 2]
```diff
@@ -51,21 +51,21 @@ func TestLogger(t *testing.T) {
 		assert.Equal(t, "[INFO]  test: this is test: who=programmer why=testing\n", rest)
 	})
 
-	t.Run("quotes values with spaces", func(t *testing.T) {
+	t.Run("drew testing formats log entries", func(t *testing.T) {
```
Comment (evanphx) [inline logger_test.go]: Probably a testing value that should be changed?

Comment (drewbailey) [reply logger_test.go]: 😳️ woops, fixed

**Pull request merged**